carbon-design-system/carbon-addons-iot-react
# feat(table): add toolbarActions to add actions to table toolbar or its overflow menu

Closes #3063 
Closes #2951 

**Summary**

You can add other actions to the toolbar or its overflow menu by using the `toolbarActions` prop on the view.toolbar. When clicking on an action it will trigger the `onApplyToolbarAction` callback passing the object used to create the action.
This object must contain an `id` and `labelText`, but it can also contain `hidden`, `disabled`, `isDelete`, `hasDivider`, or `renderIcon`, `isOverflow`, or `isActive` items.

**Change List (commits, features, bugs, etc)**

- add `toolbarActions` to `view.toolbar` table prop
- pass these actions to the TableToolbar using the same overflow menu used for aggregations (using `isOverflow`) or put them in the toolbar when not passing (`isOverflow`).
- if aggregations are enabled, any `toolbarActions` using `isOverflow` will be appended to the overflow menu
- if aggregations are not enabled, the `toolbarActions` using `isOverflow` will create the overflowMenu
- Any other `toolbarActions` not using `isOverflow` will be placed to the left of the overflow menu.

**Acceptance Test (how to verify the PR)**

- check the [basic dumb table story](https://deploy-preview-3076--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-table--basic-dumb-table) and turn on aggregations
- confirm the toolbar overflow menu contains other items.
- Edit should be disabled
- Delete should be red when hovering
- there should not be one named `Hidden` even though it's in the actions.
- when clicking on `Toggle something`
- confirm the `onApplyToolbarAction` action is triggered in storybook
- https://deploy-preview-3076--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-table--with-filters
- should show the "eye-off" icon in the toolbar and trigger the `onApplyToolbarAction` action in storybook when clicked
**Regression Test (how to make sure this PR doesn't break old functionality)**

- confirm this doesn't adversely affect the aggregations in any way.



**Things to look for during review**
